components/tabs/src/storage.rs

```diff
@@ -293,18 +295,40 @@ impl TabsStorage {
                 return crts;
             }
         };
+        // building this in to a hashset seems a little over the top?
```

Comment:
I think so, too! I'd almost suggest a sorted `Vec` (with `ORDER BY device_id`, or sorting in Rust) that you can `binary_search`, but even that feels a bit excessive 😅